### PR TITLE
Feature/con 237 api endpoint

### PR DESCRIPTION
**Pull request by iceraluk**
- Added /document-upload endpoint
- Added ability to send a file and file_path
- Added ability to send type and size validation parameters
- Added validations
- Added some token based authentication
- Added configuration for Govuk Paas
- Updated Readme file

### CODE REVIEW

File: spec/factories/client.rb [hunk 1]
```diff
@@ -0,0 +1,6 @@
+FactoryBot.define do
+  factory :client do
+    source_app { "my_app" }
+    api_key { "RbZHfHtD1h9XZvs4fGPJUgtt" }
```
Comment (karishma-ccs) [inline spec/factories/client.rb:4]: Should this be here?

File: db/schema.rb [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+  create_table "clients", id: :uuid, default: -> { "gen_random_uuid()" }, force: :cascade do |t|
+    t.string "source_app", limit: 20
+    t.string "api_key", limit: 255
+    t.datetime "created_at", precision: 6, null: false
+    t.datetime "updated_at", precision: 6, null: false
+  end
+
+  create_table "documents", id: :uuid, default: -> { "gen_random_uuid()" }, force: :cascade do |t|
+    t.string "source_app", limit: 20
```
Comment (karishma-ccs) [inline db/schema.rb:28]: Is this a foreign key?

File: app/uploaders/document_file_uploader.rb [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+
+  # Override the directory where uploaded files will be stored.
+  # This is a sensible default for uploaders that are meant to be mounted:
+  def store_dir
+    "uploads/#{model.class.to_s.underscore}/#{mounted_as}/#{model.id}"
+  end
+
+  # Add a white list of extensions which are allowed to be uploaded.
+  # For images you might use something like this:
+  def extension_whitelist
```
Comment (karishma-ccs) [inline app/uploaders/document_file_uploader.rb:22]: should we rename 'whitelist' ?

**Review by karishma-ccs — APPROVED** [review]
I made a few comments, although happy to pre-approve to not block you.

File: app/uploaders/document_file_uploader.rb (hunk 3 above)
Comment (iceraluk) [reply app/uploaders/document_file_uploader.rb:22]: this is carrierwave's default and cannot rename it

File: db/schema.rb (hunk 2 above)
Comment (iceraluk) [reply db/schema.rb:28]: no, the foreign key of the Client is on the client_id of the UncheckedDocument. We could actually remove this attribute here as it's a duplicate

File: spec/factories/client.rb (hunk 1 above)
Comment (iceraluk) [reply spec/factories/client.rb:4]: yes, this is just stubbing for testing purposes

**Pull request merged**